### DISCUSSION TITLE
Separate you apps into different repositories

### DISCUSSION THREAD

**Discussion by yuseferi** [ente-io/ente#3628] [Enhancements]
I know that your first idea was about Ente, and after that, you start Ente auth, and because it has some packages in common with Ente, you keep it into one mono-repo, but if you want to make auth successful, I recommend separating them.
I see Ente as very promising. I'm in the stage of finding a cross-platform healthy alternative, and so far, I have found Ente Auth to be one of the best alternatives.

**Comment by dnred** [comment]
What benefits would seperating them have?

So what it isn't or won't be successful if it's a monorepo? Generally people don't look at the stack all that often, they just use the apps. And even if they did what difference does seperating them make?

It is a monorepo because Ente apps share the same server.

**Reply by yuseferi** [reply]
First,  it is an open-source product. If people want to collaborate, does not need to know any other thing about other services in the project; just focus on the Ente Auth project, and it causes higher contributions,
Second, Stars, this Repo has 15k starts, but how much of it is for Ente Photo and How much is for Ente Authy, 

as just a GitHub user who likes open source and sometimes contributing, I'm very interested in the Ente Auth and have no interest in Ente Photo,  that would be great if you separate these services, I know at the beginning it would be very hard, but if you want to make it promising and successful as an open source, it is better to be a standalone project.

I see Ente Auth as a very promising alternative to Authy, and I would like to collaborate on it.

**Reply by Xyz00777** [reply]
’d like to strongly advocate for having independent apps, as they offer significant advantages in terms of security, usability, resource efficiency, and overall clarity.

1. Security Benefits: Independent apps reduce the attack surface by isolating functionalities. This makes it harder for vulnerabilities in one feature to compromise the entire system, leading to a much safer overall experience.
2. Development and User Experience: Independent apps are easier to develop and maintain, as they focus on a specific purpose without being overloaded with unnecessary features. This simplicity translates to better usability for users who only need a single, focused solution.
3. Resource Efficiency: A modular approach allows users to install only what they actually need, reducing unnecessary resource usage. Lightweight apps are not only more efficient but also provide a better experience on low-powered devices.
4. Clearer Documentation: Documentation for independent apps is often more straightforward and easier to understand because each app has its own specific scope. This separation avoids confusion and makes it easier for users and developers alike to find the information they need.
5. Simplified Installation Instructions: With independent apps, installation and setup processes can be streamlined and more user-friendly. Step-by-step instructions for a single app are far easier to follow than those for a complex, all-in-one package. This clarity lowers the entry barrier and encourages broader adoption.

For many, the inability to install and use only the features they need can be a dealbreaker. Supporting independent apps would address this and make the system more appealing to a wider audience.

**Comment by MulverineX** [comment]
I'd like this so I can self-host only Ente Auth, I don't need to host Ente Photos

**Comment by Astremy** [comment]
I would love to self-host the service I'm interrested in, without the bother of also having to setup other stuff. Plus, as said on other comments, it could improve accessibility/readability.

There are also issues showing that a monolitic solution can already be a problem due to the number of "parts" needed to actually make it work (eg. https://github.com/ente-io/ente/discussions/3767).

I know it would be a pretty huge load at the start, but I just wonder if it could be among your future plans to improve usage. It would also prevent "infinite" grow of this repository, to a point where it would no longer be really possible to self-host, nor separate.

**Comment by brog-io** [comment]
They used to have separate repos for their services. You can read on their expierence here: https://ente.io/blog/monorepo-retrospective/

**Reply by MulverineX** [reply]
Some way to more easily self-host individual components would still be really nice

**Comment by DanteMS** [comment]
Museum barely consumes any resources. It's very lightweight. 
Just use what you want and skip the rest. 
I like that I don't have to spin new containers whenever Ente gets new services.

**Reply by Xyz00777** [reply]
beside of that this is possibility yes, this is no good advisory who should be given when you care about security...
